modules/convert/convert.go

```diff
@@ -414,3 +415,31 @@ func ToLFSLock(l *git_model.LFSLock) *api.LFSLock {
 		},
 	}
 }
+
+// ToChangedFile convert a gitdiff.DiffFile to api.ChangedFile
+func ToChangedFile(f *gitdiff.DiffFile, repo *repo_model.Repository, commit string) *api.ChangedFile {
+	status := "changed"
+	if f.IsDeleted {
+		status = "deleted"
+	} else if f.IsCreated {
+		status = "added"
+	} else if f.IsRenamed && f.Type == gitdiff.DiffFileCopy {
+		status = "copied"
+	} else if f.IsRenamed && f.Type == gitdiff.DiffFileRename {
+		status = "renamed"
+	}
+
+	file := &api.ChangedFile{
+		Filename:  f.Name,
+		Status:    status,
+		Additions: f.Addition,
+		Deletions: f.Deletion,
+	}
+
+	if f.Name != "" {
+		file.HTMLURL = fmt.Sprint(repo.HTMLURL(), "/src/commit/", commit, "/", util.PathEscapeSegments(f.Name))
+		file.ContentsURL = fmt.Sprint(repo.HTMLURL(), "/raw/commit/", commit, "/", util.PathEscapeSegments(f.Name))
+	}
```

Comment:
One more thing, as GitHub does, the `deleted`(`removed`) file's name is also reported here.


```
curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/wxiaoguang/playground/pulls/1/files
[
  {
    "sha": "04d1ca8c2c853cf81b2012cc6cd232bc69caa136",
    "filename": "test-utf8.jl",
    "status": "removed",
    "additions": 0,
    "deletions": 2,
    "changes": 2,
    "blob_url": "https://github.com/wxiaoguang/playground/blob/285c45d3fd48ab98dbbecac2374c8181e235e2a8/test-utf8.jl",
    "raw_url": "https://github.com/wxiaoguang/playground/raw/285c45d3fd48ab98dbbecac2374c8181e235e2a8/test-utf8.jl",
    "contents_url": "https://api.github.com/repos/wxiaoguang/playground/contents/test-utf8.jl?ref=285c45d3fd48ab98dbbecac2374c8181e235e2a8",
    "patch": "@@ -1,2 +0,0 @@\n-ṡ_b\n-s̄_b"
  }
]
```